libs/coin-modules/coin-polkadot/src/network/sidecar.ts

```diff
@@ -655,3 +655,25 @@ export const getTransactionMaterialWithMetadata = makeLRUCache(
   () => "polkadot",
   hours(1),
 );
+
+/**
+ * Retrieve shortened metadata for a transaction
+ *
+ * @async
+ * @param {string} transaction - the transaction to shorten
+ *
+ * @returns {string>} - the shortened metadata for a transaction
+ */
+export const shortenMetadata = async (transaction: string): Promise<string> => {
```

Comment:
It should be accessible through `PolkadotAPI` class.
Also, it doesn't seems to be part of sidecar API 🤔

Reply:
Good point.
I agree that this is not be part of side car API in theory.
But from the ticket
https://ledgerhq.atlassian.net/browse/BACK-7065

The backend team wants add an endpoint 
https://polkadot-sidecar.coin.ledger.com/runtime/metadata/
for that.

So I think I may wait for the endpoint from backend team and decide whether put this method in sidecar.

Reply:
I think we could define this function in another file (to reflect accurately what is in sidecar), and decorate this function in `PolkadotAPI`.

Reply:
Sorry I did not know how the script worked in details, if it's just a input => output endpoint, we can adapt the ticket and just provide the service. I found it fine if we had runtime/metadata/shortened taking usual metadata and shortening it but we want to drop the fork specific features so we may need to think again about this

Reply:
@hedi-edelbloute 
yes, it is just a input => output endpoint.

for reference, here is the metadate shortener endpoint doc:
https://github.com/Zondax/ledger-polkadot-generic-api/blob/master/docs/openapi.yaml#L63

As you can see my implementation and the doc, we don't even need to retrieve the full runtime metadata.